modules/ztp-image-based-upgrade-procedure-rollback.adoc

```diff
@@ -0,0 +1,130 @@
+// * edge_computing/image-based-upgrade/ztp-image-based-upgrade.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="ztp-image-based-upgrade-procedure-rollback_{context}"]
+= Rolling back an image-based upgrade on managed clusters at scale
+
+Roll back the changes on a set of managed clusters if you encounter unresolvable issues after an upgrade.
+You need to create a separate `ImageBasedGroupUpgrade` CR and define the set of managed clusters that you want to roll back.
+
+[IMPORTANT]
+====
+Ensure that the action combinations created by appending actions are supported.
+For more information, see _Supported action combinations_.
+====
+
+.Prerequisites
+
+* Log in to the hub cluster as a user with `cluster-admin` privileges.
+
+.Procedure
+
+. Check the status of the upgrade by running the following command:
++
+--
+[source,terminal]
+----
+$ oc get ibgu -o yaml
+----
+
+.Example output
+[source,yaml]
+----
+[...]
+status:
+  clusters:
+  - completedActions:
+    - action: Prep
+    name: spoke1
+  - completedActions:
+    - action: Prep
+    failedActions:
+    - action: Upgrade
+    name: spoke4
+  - failedActions:
+    - action: Prep
+    name: spoke6
+[...]
+----
+--
+
+. Create a YAML file on the hub cluster that contains the `ImageBasedGroupUpgrade` CR, for example, named `example-group-rollback.yaml`:
++
+[source,yaml]
+----
+apiVersion: lcm.openshift.io/v1alpha1
+kind: ImageBasedGroupUpgrade
+metadata:
+  name: example-group-rollback
+  namespace: default
+spec:
+  clusterLabelSelectors:
+  - matchExpressions:
+    - key: name
+      operator: In
+      values:
+      - spoke4
+  ibuSpec:
+    seedImageRef:
+      image: quay.io/seed/image:4.16.0-rc.1
+      version: 4.16.0-rc.1
+      pullSecretRef:
+        name: "<seed_pull_secret>"
+  extraManifests:
+    - name: example-extra-manifests
+      namespace: openshift-lifecycle-agent
+  oadpContent:
+    - name: oadp-cm
+      namespace: openshift-adp
+  plan:
+  - actions: ["Rollback", "FinalizeRollback"]
+    rolloutStrategy:
+      maxConcurrency: 200
+      timeout: 2400
+----
+
+. Apply the `example-group-rollback.yaml` file by running the following command on the hub cluster:
++
+--
+[source,terminal]
+----
+$ oc apply -f example-group-rollback.yaml
+----
+
+All managed clusters that match the defined labels are moved back to the `Rollback` and then the `Idle` stages to finalize the rollback. 
+--
+
+.Verification
+
+. Monitor the status updates by running the following command:
++
+--
+[source,terminal]
+----
+$ oc get ibgu -o yaml
+----
+//If it's a separate ibgu, do we need to specify which CR we want to see?
+
+.Example output
+[source,yaml]
+----
+[...]
+status:
+  clusters:
+  - completedActions:
+    - action: Prep
+    name: spoke1
+  - completedActions:
+    - action: Prep
+    - action: Rollback
+    - action: FinalizeRollback
+    failedActions:
+    - action: Upgrade
+    name: spoke4
+  - failedActions:
+    - action: Prep
+    name: spoke6
```

Comment:
this is not accurate, rollback only applies to successful upgrades. Also since we are creating a separate ibgu for rollback the previous actions won't be in the status.

Reply:
Then the verification status should be updated as well. Does this look good for Rollback with a separate IBGU?
```status:
  clusters:
  - completedActions:
    - action: Rollback
    - action: FinalizeRollback
    name: spoke4